app/src/main/java/com/guard/security/crc/crc_guard_app/webview/ManagerWebClient.java

```diff
@@ -55,18 +78,127 @@ public void run() {
-            gvContext.startActivity(intent);
-        }else{
+           // Log.i("PRUEBA",error.getDescription().toString());
+            if (error.getDescription().toString().equals("net::ERR_CACHE_MISS")) {
+                Intent intent = new Intent(gvContext, MainActivity.class);
+                gvContext.startActivity(intent);
+            }else{
+                Intent intent = new Intent(gvContext, LocalHomeActivity.class);
+                gvContext.startActivity(intent);
+            }
+            //Posibles erroes:
+            /*
+            "net::ERR_CONNECTION_REFUSED"
+            "net::ERR_INTERNET_DISCONNECTED"
+            "net::ERR_CONNECTION_ABORTED"
+             */
+            /*
+            if (error.getDescription().toString().equals("net::ERR_CONNECTION_ABORTED")) {
+                Intent intent = new Intent(gvContext, LocalHomeActivity.class);
+                gvContext.startActivity(intent);
+            }
+            if (error.getDescription().toString().equals("net::ERR_INTERNET_DISCONNECTED")) {
+                Intent intent = new Intent(gvContext, LocalHomeActivity.class);
+                gvContext.startActivity(intent);
+            }
+            if (error.getDescription().toString().equals("net::ERR_CONNECTION_REFUSED")) {
+                Intent intent = new Intent(gvContext, LocalHomeActivity.class);
+                gvContext.startActivity(intent);
+            }*/
+
+        } else {
             view.setVisibility(View.INVISIBLE);
             new ErrorController(gvContext).showErrorDialog();
         }
+    }
+
+    @Override
+    public boolean shouldOverrideUrlLoading(WebView v, String u) {
+        v.loadUrl(u);
+        if (ContextCompat.checkSelfPermission(this.gvContext,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
+        }
+        v.setDownloadListener(this);
+        return true;
+    }
+
+    @Override
+    public void onDownloadStart(String url, String userAgent,
+                                String contentDisposition, String mimeType,
+                                long contentLength) {
+
+
+        DownloadManager.Request request = new DownloadManager.Request(
+                Uri.parse(url));
+        //Se procesa el contenido para buscar el nombre completo de la aplicacion
+        //Ya que si se hace con URLUtil.guessFileName() se obtiene un nombre como "f.apk" o "f.bin"
+        String[] Nombre = contentDisposition.split(";");
+        String FNombre = "";
+        for (int i = 0; i < Nombre.length; i++) {
+            if (Nombre[i].contains("=")) {
+
+                String[] File = Nombre[i].split("=");
+                for (int f = 0; f < File.length; f++) {
+
+                    if (!File[f].contains("filename")) {
+                        FNombre = File[f].replace("\"", "");
+                        break;
+                    }
+                }
+            }
+            if (FNombre != "") {
```

Comment:
Our static analysis engine detects that your pull request includes a change from a Comparison of String objects using `==` or `!=`. According to [this description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq), unless both strings are either constants in a source file, or have been interned using the `String.intern()` method, the same string value may be represented by two different String objects. Consider using the `equals(Object)` method instead.